ui/lib/sync/addon/components/secrets/page/destinations/destination/secrets.ts

```diff
@@ -50,13 +48,22 @@ export default class SyncSecretsDestinationsPageComponent extends Component<Args
   }
 
   @action
-  async update(association: SyncAssociationModel, operation: string) {
+  async update(association: AssociatedSecret, operation: string) {
     try {
-      await association.save({ adapterOptions: { action: operation } });
+      const { name, type } = this.args.destination;
+      const { mount, secretName } = association;
+      const body = { mount, secretName };
+
+      if (operation === 'set') {
+        await this.api.sys.systemWriteSyncDestinationsTypeNameAssociationsSet(name, type, body);
+      } else {
+        await this.api.sys.systemWriteSyncDestinationsTypeNameAssociationsRemove(name, type, body);
+      }
```

Comment:
Surprised you didn't do the Jordan special 😂
```js
const method = capitalize(operation)
`systemWriteSyncDestinationsTypeNameAssociations${method}`
```
(i'm just kidding, I think what you have here makes sense!)

Reply:
😆 believe me I thought about it!